Source/CarthageKit/Project.swift

```diff
@@ -371,7 +371,43 @@ public final class Project { // swiftlint:disable:this type_body_length
 					}
 					.map { $0.0.name }
 					.collect()
+		}
+	}
+
+	/// Finds the required dependencies and their corresponding version specifiers for each dependency in Cartfile.resolved.
+	func requirementsByDependency(
+		resolvedCartfile: ResolvedCartfile,
+		tryCheckoutDirectory: Bool
+	) -> SignalProducer<[Dependency: [Dependency: VersionSpecifier]], CarthageError> {
+		return SignalProducer(value: resolvedCartfile)
+			.map { resolvedCartfile -> [(Dependency, PinnedVersion)] in
+				return resolvedCartfile.dependencies
+					.map { k, v in (k, v) }
 			}
+			.flatMap(.merge) { dependencies -> SignalProducer<[Dependency: [Dependency: VersionSpecifier]], CarthageError> in
+				return SignalProducer(dependencies)
+					.flatMap(.merge) { dependencyAndPinnedVersion -> SignalProducer<(Dependency, (Dependency, VersionSpecifier)), CarthageError> in
```

Comment:
We shouldn't use the `merge` strategy here. If you have 50 dependencies, this would try to process all 50 at once. Since this is reading from disk, that could result in thrashing. Elsewhere we use `.concurrent(4)`; I think that'd be appropriate here.